graphics/blender/Portfile

```diff
@@ -145,6 +151,55 @@ post-patch {
     reinplace -E {s|(include )"(openjpeg.h)"|\1<openjpeg-2.3/\2>|} \
         ${worksrcpath}/source/blender/imbuf/intern/jp2.c
 
+    # Make it so that CMake is able to find the MacPorts USD library
+    if {![variant_isset usd_monolithic]} {
+        # When Blender is configured to build USD as an internal
+        # dependency, Blender sets USD's compile options so that it
+        # builds as a single monolithic library. However, the default
+        # variant of the MacPorts USD port builds the USD libraries as
+        # modular individual libraries, so in that case we make it so
+        # that Blender is able to see the individual USD modules.
+        reinplace -E {s/(usd_m usd_ms)/usd \1/} \
+            ${worksrcpath}/build_files/cmake/Modules/FindUSD.cmake
+        reinplace "/blender_add_lib/i\\
+list(APPEND USD_LIBRARIES\\
+\\  gf plug sdf tf vt\\
+\\  usdGeom usdLux usdShade usdUtils\\
+)\\
+\\
+" \
+            ${worksrcpath}/source/blender/io/usd/CMakeLists.txt
+    }
+    foreach f [list \
+        ${worksrcpath}/source/blender/io/usd/CMakeLists.txt \
+        ${worksrcpath}/tests/gtests/usd/CMakeLists.txt \
+    ] {
+        reinplace {/add_definitions.*DPXR_STATIC/s/^/#/g} $f
+    }
+
+    if {![variant_isset embree_static]} {
+        # Remove these items from the list of Embree components that
+        # Blender's FindEmbree.cmake script will try to find. The
+        # default variant of the MacPorts Embree port does not generate
+        # individual libraries with these names, because Embree's
+        # default compile options build it as a dynamic library. The
+        # individual libraries only get generated when building Embree
+        # as a static library.
+        foreach re [list \
+            {/embree_avx/d} \
+            {/embree_avx2/d} \
+            {/embree_sse42/d} \
+            {/lexers/d} \
+            {/math/d} \
+            {/simd/d} \
+            {/sys/d} \
+            {/tasking/d} \
+        ] {
+            reinplace $re \
+                ${worksrcpath}/build_files/cmake/Modules/FindEmbree.cmake
+        }
```

Comment:
Your continued use of reinplaces where patchfiles are preferred is frustrating. Each of these reinplaces, for example, removes just a single line from the file, and all of the lines are together. It would be shorter, simpler, more clear what you're doing to use a patchfile instead, and it would prevent problems in the future. Today, for example, the file doesn't contain any other lines that contain the three characters "sys" in that order, but you cannot be certain that the file will not be modified in the future to contain such other occurrences or that removing such other occurrences would not have negative effects on the build.

Reply:
And I find it frustrating that you seem to keep implying that patchfiles are always preferred over reinplaces. If Blender's source code was relatively unchanging, then I might agree with you, but Blender is currently undergoing a period of rapid change and development. There's no guarantee that the entire reinplace won't simply disappear a couple of updates from now.

I will admit that the regexes, as I have written them, are a bit inefficient and have the possibility for ambiguity later, but that simply means that the regexes should be modified to make them better... it doesn't necessarily mean that they shouldn't be regexes at all, in favor of a patchfile. Conversely to what you said, one could argue that you can't be certain that the context lines before and after a patch block won't be modified in the future such that the patch will fail. And I would argue that a project like Blender, where the code is currently in rapid flux, that this situation is more likely to happen, whereas a well-written regular expression, which is completely position independent, would actually be _less_ likely to have problems.

Reply:
I've modified the regexes to be more specific.